ddtrace/contrib/requests/connection.py

```diff
@@ -29,10 +28,14 @@ def _wrap_send(func, instance, args, kwargs):
     if not request:
         return func(*args, **kwargs)
 
-    parsed_uri = parse.urlparse(request.url)
-    hostname = parsed_uri.hostname
-    if parsed_uri.port:
-        hostname = "{}:{}".format(hostname, parsed_uri.port)
+    url = request.url
+    i = url.index("://") + 3
+    try:
+        j = url.index("/", i)
```

Comment:
`http://example.com?foo` and `http://example.com#foo` are valid URLs. We probably need to handle them too.

Reply:
Indeed! Good catch!